File: API/src/main/java/br/com/pizzariaapi/API/Entity/Item.java
package br.com.pizzariaapi.api.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;


    @Entity
    @Getter
    @Setter
    @Table(name = "tb_item")
    public class Item {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false, unique = true)
        private Long id;
        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "pedido_id")
        @JsonIgnore
        private Pedido pedido;
        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "produto_id")
        private Produto produto;
        @Column(name = "quantidade")
        private int quantidade;
    }

File: Client/src/app/MODEL/item-model/item.ts
import { Atributo } from "../atributo-model/atributo";
import { Pedido } from "../pedido-model/pedido";
import { Produto } from "../produto-model/produto";
import { Sabor } from "../sabor-model/sabor";


export class Item {
    id!: number;
    produto!: Produto; 
    quantidade!: number;
    sabors!: Sabor[]; 
    atributoEspecifico!: Atributo;
    pedido!:Pedido
}
